tests/components/sunweg/test_sensor.py

```diff
@@ -0,0 +1,191 @@
+"""Tests for the Sun WEG sensor."""
+
+from datetime import timedelta
+
+from freezegun.api import FrozenDateTimeFactory
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.sunweg.const import DOMAIN
+from homeassistant.components.sunweg.coordinator import SCAN_INTERVAL
+from homeassistant.core import HomeAssistant, dt_util
+from homeassistant.setup import async_setup_component
+
+from .common import SUNWEG_MOCK_ENTRY
+
+from tests.common import async_fire_time_changed
+
+
+async def test_sensor_total(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    freezer: FrozenDateTimeFactory,
+    api_fixture,
+    plant_fixture_alternative,
+) -> None:
+    """Test sensor type total."""
+    mock_entry = SUNWEG_MOCK_ENTRY
+    mock_entry.add_to_hass(hass)
+
+    assert await async_setup_component(hass, DOMAIN, mock_entry.data)
+    freezer.tick(SCAN_INTERVAL)
+    async_fire_time_changed(hass, dt_util.now() + timedelta(minutes=20))
```

Comment:
```suggestion
    async_fire_time_changed(hass)
```
You don't need to move the time forwards if you've done the `freezer.tick(SCAN_INTERVAL) as it's already moved.  Apologies this might be my bad as I pasted a bad example when I pointed you to the freezer.  I think after you fix this and the other instances of it that'll be it from me.